src/Compilers/Core/Rebuild/CompilationOptionsReader.cs

```diff
@@ -220,8 +233,8 @@ public ImmutableArray<MetadataReferenceInfo> GetMetadataReferences()
             using (stream)
             {
                 // todo: IVT and EncodedStringText.Create?
-                var embeddedText = SourceText.From(stream, encoding: encoding, checksumAlgorithm: hashAlgorithm, canBeEmbedded: true);
-                return (embeddedText, compressedHash);
+                var embeddedText = SourceText.From(stream, encoding: sourceTextInfo.SourceTextEncoding, checksumAlgorithm: sourceTextInfo.HashAlgorithm, canBeEmbedded: true);
+                return new EmbeddedSourceTextInfo(sourceTextInfo, embeddedText, compressedHash is object ? compressedHash.ToImmutableArray() : ImmutableArray<byte>.Empty);
```

Comment:
nit: `compressedHash?.ToImmutableArray() ?? ImmutableArray<byte>.Empty` does the same thing here

Reply:
i have a lot of `?` about this suggestion 😉